app/main/util/search_utils.py

```diff
@@ -7,21 +7,41 @@
 
 
 def format_opensearch_results(results):
-    results_clone = results
+    """Format date fields of the _source object inside results"""
+    results_clone = results.copy()
     for result in results_clone:
         for key, value in result["_source"].items():
             if "date" in key:
                 result["_source"][key] = format_opensearch_date(value or "")
     return results_clone
 
 
+def filter_opensearch_highlight_results(results):
+    """Filter highlight results for fields that are not needed"""
+    results_clone = results.copy()
+    for result in results_clone:
+        if result.get("highlight", None):
+            keys_to_remove = [
+                key for key in result["highlight"].keys() if ".keyword" in key
```

Comment:
what keys might we be removing here? i.e what keys use .keyword? I understand we donyt want transferring body but would we not want some of them?

Reply:
@anthonyhashemi did some more reading up and seems like they might be important, though in our case we get duplicate results like this, I'm not entirely sure if we want this or not
<img width="685" alt="image" src="https://github.com/user-attachments/assets/c5f95fd1-36ec-4ef9-8e4d-c9491e98a15c">

Reply:
let's hop on a call to investigate this together

Reply:
called and decided its fine to remove all of the .keywords for now as always will have a text equivalent (field without the .keyword mapping) match but have created a ticket to investigate whether we can utuilise keyword for optimizations or remove it from index.